File: src/main/java/net/treset/minecraftlauncher/util/ui/sort/InstanceDetailsNameComparator.java
package net.treset.minecraftlauncher.util.ui.sort;

import net.treset.minecraftlauncher.LauncherApplication;
import net.treset.minecraftlauncher.data.InstanceData;

import java.util.Comparator;

public class InstanceDetailsNameComparator implements Comparator<InstanceData> {
    @Override
    public int compare(InstanceData e1, InstanceData e2) {
        return e1.getInstance().getKey().getName().compareTo(e2.getInstance().getKey().getName());
    }

    @Override
    public String toString() {
        return LauncherApplication.stringLocalizer.get("sort.name");
    }
}

File: client/composeApp/src/commonMain/kotlin/dev/treset/treelauncher/backend/util/sort/InstanceDetailsNameComparator.kt
package dev.treset.treelauncher.backend.util.sort

import dev.treset.treelauncher.backend.data.InstanceData
import dev.treset.treelauncher.localization.Strings

class InstanceDetailsNameComparator : Comparator<InstanceData> {
    override fun compare(e1: InstanceData, e2: InstanceData): Int {
        return e1.instance.name.value.lowercase().compareTo(e2.instance.name.value.lowercase())
    }

    override fun toString(): String = Strings.sortBox.sort.name()
}
